src/ape/api/accounts.py

```diff
@@ -135,8 +142,19 @@ def transfer(
             txn.data = self._convert(data, bytes)
 
         if value:
+            if "send_everything" in kwargs and kwargs["send_everything"]:
+                raise ValueError(
+                    "Kwarg send_everything=True requires transfer without value argument"
+                )
             txn.value = self._convert(value, int)
 
+        if not value:
+            if not kwargs.get("send_everything"):
+                raise ValueError(
+                    "Transfer without value argument requires kwarg send_everything=True"
+                )
+            return self.call(txn, kwargs["send_everything"])
+
         return self.call(txn, send_everything=value is None)
```

Comment:
A little more streamlined here
```suggestion
            txn.value = self._convert(value, int)
            return self.call(txn, send_everything=value is None)

        elif not kwargs.get("send_everything"):
            raise ValueError(
                "Transfer without value argument requires kwarg send_everything=True"
            )
        else:
            return self.call(txn, send_everything=True)

```

Reply:
The nitpick: having nested if statements can be harder to read than trying to flatten them down to one level of `if..elif..else` statements, which makes it easier to reason about